kvbc/src/v4blockchain/detail/blockchain.cpp

```diff
@@ -36,7 +36,8 @@ Blockchain::Blockchain(const std::shared_ptr<concord::storage::rocksdb::NativeCl
   if (genesis_blockId) {
     ConcordAssertGT(genesis_blockId.value(), INVALID_BLOCK_ID);
     ConcordAssertLE(genesis_blockId.value(), last_reachable_block_id_);
-    ConcordAssert(!hasBlock(genesis_blockId.value() - 1));
+    // The following is not necessarily true when we use DeleteFilesInRange()
+    // ConcordAssert(!hasBlock(genesis_blockId.value() - 1));
```

Comment:
I guess it's related to the note in the doc ```Leaking keys on the boundaries (logically and physically) - but it should have no impact on the product.``` right?

It seems that the recent introduction of saving the genesis id as a KV , complements this behavior and prevents deviation across replicas.

Reply:
Can it fail Unit tests? Where a block is assumed to be deleted? 
Did we think if it can cause deviation across replicas and whether those deviations are harmful?

Reply:
Apparently it does not fail unit tests.
It will cause deviation across replicas about keys which are not supposed to exist. The logical keys leaking are not supposed to be referenced any more, so it will normally not impact state transfer, block hashes, merkle trees, etc.